crabs/detection_tracking/train_model.py

```diff
@@ -116,7 +118,67 @@ def setup_trainer(self):
             limit_train_batches=self.limit_train_batches,
         )
 
-    def train_model(self):
+    def optuna_maximise_val_precision_recall(
+        self, trial: optuna.Trial
+    ) -> float:
+        """Objective function for Optuna.
+
+        When used with Optuna, it wil maximise precision and recall on the
+        validation set.
+
+        Parameters
+        ----------
+        trial : optuna.Trial
+            The trial to optimise.
+
+        Returns
+        -------
+        float
+            The value to maximise.
+        """
+        # Sample hyperparameters from the search space for this trial
+        # learning rate (a float)
+        if "learning_rate" in self.config["optuna"]:
```

Comment:
It would be nice if we could be flexible for any hyperparameter defined in `self.config['optuna']`.

So we could say: if the hyperparameter is an integer, use `trial.suggest_int(..)`. But if it is a float, use `trial.suggest_float(..)`.

Can we do this?